test/suites/smoke/test-staking-rewards.ts

```diff
@@ -582,7 +583,12 @@ describeSuite({
         ).round.ideal;
         const idealIssuance = new Perbill(idealInflation).of(totalIssuance);
 
-        totalRoundIssuance = roundDuration.mul(idealIssuance).div(idealDuration);
+        totalRoundIssuance = roundDuration
```

Comment:
I don't undeerstand why this is needed. Do they also increase the precision by 1000 in the rust code ? Because both are doing integer computation so it should match

Reply:
The Rust code doesn't augment the precision, but since I ignore the details of implementation of BN.js, my guess is that there is a discrepancy there. In this case augmenting to one or two decimal places I think would suffice, but I wanted to be on the safe side

Reply:
btw I added the numerical reasoning behind to the desc

Reply:
I would prefer we don't change the precision but instead we write the correct computation. When dealing with percentage split of a integer, it is usually either rounding the first part of the second part (which can be emulated by computing the 2nd part first).
Ex1: A = value * P / 100; B = value - A
Ex2: B = value * (100-P) / 100; A = value - B

Reply:
Okay I see the point. But to me both solutions seem to be manipulating the rounding at some point, either by augmenting precision or purposely looking at the other segment of the percent. Anyway I'll try yours and see if it works as expected :)

Reply:
Can you confirm that the rounding issue does not come from `reservedReward.eq(reservedInflation)`.

It should be rounding to the nearest:

```rs
assert_eq!(Percent::from_percent(34) * 10u64, 3);
assert_eq!(Percent::from_percent(36) * 10u64, 4);
```

Reply:
Checked and I was able to reproduce one case where it might be a rounding issue coming from the `reservedPercentage.of(totalRoundIssuance)` landing on a 5.

Following @crystalin comment and calculating it as the following seems to fix the issue!

```
const remainingPercent = new Percent(new BN(100).sub(percentage))
const remainingReward = remainingPercent.of(totalRoundIssuance);
const reservedReward = totalRoundIssuance.sub(remainingReward);
```

Reply:
If you follow what I said, you can make it in 2 lines of code instead of 3 :p

Reply:
> Can you confirm that the rounding issue does not come from `reservedReward.eq(reservedInflation)` or `new Perbill(idealInflation).of(totalIssuance)`.
> 
> It should be rounding to the nearest:
> 
> ```rust
> assert_eq!(Percent::from_percent(34) * 10u64, 3);
> assert_eq!(Percent::from_percent(36) * 10u64, 4);
> ```

Both Percent and Perbill are rounding as expected

Reply:
Had a look at this issue today, and I believe commit https://github.com/moonbeam-foundation/moonbeam/pull/3065/commits/6e67888e2e5baa6cc31fe1c8632ad856b901935d will fix it.

The issue was that we were not calculating the inflation distribution individually, instead, we were summing up all the percentages and applying the aggregated percentage at the end, which would cause the rounding issue for some rounds.

Reply:
Nice catch Rodri ! I wasn't aware of that. I will approve